homeassistant/components/lutron/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow to configure the Lutron integration."""
+from __future__ import annotations
+
+import logging
+from urllib.error import HTTPError
+
+from pylutron import Lutron
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LutronConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User prompt for Main Repeater configuration information."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """First step in the config flow."""
+        errors = {}
+
+        if user_input is not None:
+            # Validate user input
+            user_input.get(CONF_USERNAME)
+            user_input.get(CONF_PASSWORD)
+            ip_address = user_input.get(CONF_HOST)
+
+            # Now that we have the data, check and see if we can connect and get a GUID
+            main_repeater = Lutron(
+                ip_address,
+                user_input.get(CONF_USERNAME),
+                user_input.get(CONF_PASSWORD),
+            )
+
+            def _load_db() -> bool:
+                main_repeater.load_xml_db()
+                return True
+
+            try:
+                await self.hass.async_add_executor_job(_load_db)
+                guid = main_repeater.guid
+            except UnknownConnectError:
+                errors["base"] = "config_errors"
+                guid = "failed"
+            except HTTPError as ex:
+                # In a future version we can get more specific with the HTTP code
+                _LOGGER.debug("Exception Type: %s", type(ex).__name__)
+                errors["base"] = "connect_error"
+                guid = "failed"
+
+            if not errors:
```

Comment:
```suggestion
            else:
```

Reply:
This is useful to be able to see clearly the repeater DB you are connected to.  It is also possible that you would connect to something but a guid wouldn't be found.  I'd like to keep this.

Reply:
This just means, if the code did not reach the except clauses

Reply:
I see the redundancy now and cleaned that up.